src/core/thread/mle.cpp

```diff
@@ -246,6 +252,7 @@ ThreadError Mle::Stop(void)
         mNetif.RemoveUnicastAddress(mLeaderAloc);
     }
 
+    mNetif.GetPendingDataset().StopDelayTimer();
```

Comment:
In general, when a device is attached to a network, it should always use the Delay Timer that was received from its Parent (i.e. the one in `mNetwork`).  If the device detaches from the network, then it should use the one it has stored locally (i.e. the one in `mLocal`).  So rather than starting the Delay Timer in `SetStateChild()` and `SetStateRouter()`, I think it's better if we just start the timer when we are detached.  Does that make sense?